### PR TITLE
Improve Gitlab Integration closes #276

### LINKED ISSUE #276: Improve Gitlab Integration
Hi @jayfk  and @samdroid-apps 
Thank you for providing a Gitlab integration in the first place we are total fans. 
The question I have is whether you accept changes to it. 

Case in point would be a flag to allow Gitlab to `merge_when_pipeline_succeeds` and another flag for `should_remove_source_branch`. We currently have a custom made script that does exactly that because we don't want to manually merge changes that pass all tests.

The problem here would be that these changes are somewhat specific to Gitlab.  I could imagine an `os.env` kind of flag.  Let me know what you think. 

Thanks,
Kai

### PR DESCRIPTION
**Pull request by kairichard**
This turned out bigger than I expected - wouldn't mind some feedback. 
If the direction worthy to continue on I would clean it up some more and than proceed with testing it in the wild.

closes #276

### CODE REVIEW

**Comment by jayfk** [comment]
Thanks for this! 

Tests seem to fail because of some unrelated pip foo, I'll take care of that over the weekend.

File: pyup/bot.py [hunk 1]
```diff
@@ -455,28 +455,17 @@ def create_pull_request(self, title, body, new_branch):
...
 
         return self.provider.create_pull_request(
             repo=self.bot_repo if self.bot_token else self.user_repo,
-            title=title,
-            body=body,
-            base_branch=self.config.branch,
-            new_branch=new_branch,
-            pr_label=self.config.label_prs,
-            assignees=self.config.assignees
+            **config
```
Comment (jayfk) [inline pyup/bot.py]: This is the only thing I'm a bit concerned about. The code for the online service is subclassing internally quite a lot and having functions with parameters helps quite a lot to fix errors.

Maybe pass provider specific code as `**extra`?

Comment (kairichard) [reply pyup/bot.py]: Thanks for the feedback - I like the `**extra` idea.

Comment (kairichard) [reply pyup/bot.py]: Will try to follow up as soon as possible.

Comment (kairichard) [reply pyup/bot.py]: So basically you want to keep the list of keyword arguments and pass `extra=self.config` additionally? Or am I missing something?

Comment (jayfk) [reply pyup/bot.py]: Exactly 👍

Comment (kairichard) [reply pyup/bot.py]: Addressed your comments in  d34f29e and 34b70d2

**Comment by jayfk** [comment]
Is this ready to be merged? :)

**Comment by kairichard** [comment]
We haven't run any real-world tests with it - although codewise it is ready. 
Once we successfully ran an (py-)update I will let you know, probably until end of this week.

**Comment by kairichard** [comment]
I verified this works as intended with `pip install -e .` and manually running it against a test-repo. Meaning we are good to from my side.

**Comment by jayfk** [comment]
Awesome, thanks a lot for your work on this!

**Pull request merged**